test/e2e/helpers.go

```diff
@@ -823,3 +829,90 @@ func getPodLogs(ctx context.Context, clientset *kubernetes.Clientset, pod corev1
 	}
 	return b.String()
 }
+
+// InstallHelmChart takes a helm repo URL, a chart name, and release name, and installs a helm release onto the E2E workload cluster
+func InstallHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, repoURL, chartName, releaseName string, values []string) {
+	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+	kubeConfigPath := clusterProxy.GetKubeconfigPath()
+	settings := helmCli.New()
+	settings.KubeConfig = kubeConfigPath
+	actionConfig := new(helmAction.Configuration)
+	err := actionConfig.Init(settings.RESTClientGetter(), "default", "secret", Logf)
+	Expect(err).To(BeNil())
+	i := helmAction.NewInstall(actionConfig)
+	i.RepoURL = repoURL
+	i.ReleaseName = releaseName
+	Eventually(func() error {
+		cp, err := i.ChartPathOptions.LocateChart(chartName, helmCli.New())
+		if err != nil {
+			return err
+		}
+		p := helmGetter.All(settings)
+		valueOpts := &helmVals.Options{}
+		valueOpts.Values = values
+		vals, err := valueOpts.MergeValues(p)
+		if err != nil {
+			return err
+		}
+		chartRequested, err := helmLoader.Load(cp)
+		if err != nil {
+			return err
+		}
+		release, err := i.RunWithContext(ctx, chartRequested, vals)
+		if err != nil {
+			return err
+		}
+		Logf(release.Info.Description)
+		return nil
+	}, input.WaitForControlPlaneIntervals...).Should(Succeed())
+}
+
+// WaitForWorkloadClusterKubeconfigSecret retries during E2E until the workload cluster kubeconfig secret exists
+func WaitForWorkloadClusterKubeconfigSecret(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput) {
+	// Ensure the workload cluster kubeconfig secret exists before getting the workload cluster clusterProxy object
+	Eventually(func() error {
+		cl := input.ClusterProxy.GetClient()
+		secret := &corev1.Secret{}
+		key := client.ObjectKey{
+			Name:      fmt.Sprintf("%s-kubeconfig", input.ConfigCluster.ClusterName),
+			Namespace: input.ConfigCluster.Namespace,
+		}
+		err := cl.Get(ctx, key, secret)
+		if err != nil {
+			return err
+		}
+		return nil
+	}, input.WaitForControlPlaneIntervals...).Should(Succeed())
+}
+
+// WaitForDaemonset retries during E2E until a daemonset's pods are all Running
+func WaitForDaemonset(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, cl client.Client, name, namespace string) {
```

Comment:
@CecileRobertMichon this is the alternative solution to validate the `cloud-node-manager` pods. Rather than comparing against number of machines, let's just make sure the daemonset pods are all running. There are several reasons out of our control why the number of nodes running these pods may not be easily predictable. Rather than try to deal with all of those edge cases, we can simply rely upon the daemonset API to pick the appropriate schedulable nodes, and ensure that the pods reach a Running state.